src/Compilers/CSharp/Test/Semantic/SourceGeneration/GeneratorDriverTests.cs

```diff
@@ -3058,5 +3058,43 @@ public void Timing_Info_Only_Includes_Last_Run()
             Assert.NotEqual(TimeSpan.Zero, generatorTiming2.ElapsedTime);
             Assert.True(generatorTiming.ElapsedTime > generatorTiming2.ElapsedTime);
         }
+
+        [Fact]
+        public void Post_Init_Trees_Are_Reparsed_When_ParseOptions_Change()
+        {
+            var source = "class C{}";
+            var postInitSource = "class D {  (int, bool) _field; }";
+
+            var parseOptions = TestOptions.RegularPreview;
+            Compilation compilation = CreateCompilation(source, options: TestOptions.DebugDll, parseOptions: parseOptions);
+            compilation.VerifyDiagnostics();
+
+            var generator = new PipelineCallbackGenerator(ctx =>
+            {
+                ctx.RegisterPostInitializationOutput(c => c.AddSource("D", postInitSource));
+            }).AsSourceGenerator();
+
+            GeneratorDriver driver = CSharpGeneratorDriver.Create(new[] { generator }, parseOptions: parseOptions, driverOptions: new GeneratorDriverOptions(IncrementalGeneratorOutputKind.None, trackIncrementalGeneratorSteps: true));
+            driver = driver.RunGeneratorsAndUpdateCompilation(compilation, out compilation, out var diagnostics);
+            Assert.Empty(diagnostics);
+
+            // change the parse options so that the tree is no longer accepted
+            parseOptions = parseOptions.WithLanguageVersion(LanguageVersion.CSharp1);
+            compilation = CreateCompilation(source, options: TestOptions.DebugDll, parseOptions: parseOptions);
+            driver = driver.WithUpdatedParseOptions(parseOptions);
+
+            // change some other options to ensure the parseOption change tracking flows correctly
+            driver = driver.AddAdditionalTexts(ImmutableArray<AdditionalText>.Empty);
+
+            driver = driver.RunGeneratorsAndUpdateCompilation(compilation, out compilation, out diagnostics);
+            diagnostics.Verify();
```

Comment:
There should be diagnostics at this stage though because the language version is csharp 1. Believe we should validate these diagonstics show up here, likely by just grabbing the diagnostics off the syntax tree in the compilation.

Reply:
These are the generator diagnostics. I'll add a check for the compilation.